plugins/notifications-backend/src/service/db.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
+
+import knex, { Knex } from 'knex';
+import merge from 'lodash/merge';
+
+// initDB
+// creates DB client and tables
+export async function initDB(dbConfig: Config): Promise<Knex<any, any>> {
+  // create db client
+  const knexConfig = merge(
```

Comment:
I really don't understand why you are doing totally new config reading here? It should be handled by the `PluginDatabaseManager`